### PR TITLE
interfaces/opengl: add support for Imagination PowerVR

### PR DESCRIPTION
**Pull request by EthanHsieh**
interfaces/opengl: add support for Imagination PowerVR

### CODE REVIEW

**Comment by mvo5** [comment]
This is needed on on stable before 2021-07-02

File: interfaces/builtin/opengl.go [hunk 1]
```diff
@@ -114,6 +114,9 @@ unix (bind,listen) type=seqpacket addr="@cuda-uvmfd-[0-9a-f]*",
 # https://github.com/Xilinx/XRT/tree/master/src/runtime_src/core/edge/drm
 /sys/devices/platform/amba{,_pl@[0-9]*}/amba{,_pl@[0-9]*}:zyxclmm_drm/* r,
 
+# Imagination PowerVR driver
+/dev/pvr_sync rw,
```
Comment (anonymouse64) [inline interfaces/builtin/opengl.go:118]: can you run `udevadm info --query property /dev/pvr_sync` and paste the output in this PR? it helps us ensure that the rule you provided for udev tagging is appropriate

**Review by anonymouse64 — COMMENTED** [review]
+1 after udevadm output is provided

File: interfaces/builtin/opengl.go (hunk 1 above)
Comment (EthanHsieh) [reply interfaces/builtin/opengl.go:118]: Sure. Here is the output of udevadm info
$ udevadm info --query property /dev/pvr_sync
DEVPATH=/devices/virtual/misc/pvr_sync
DEVNAME=/dev/pvr_sync
MAJOR=10
MINOR=61
SUBSYSTEM=misc
USEC_INITIALIZED=14533846
TAGS=:snap_ubuntu-core-kiosk_hook_connect-plug-graphics-core20:snap_ubuntu-core-kiosk_ubuntu-core-kiosk:snap_ubuntu-core-kiosk_hook_install:snap_ubuntu-core-kiosk_dae
mon:snap_ubuntu-core-kiosk_hook_post-refresh:snap_ubuntu-core-kiosk_hook_configure:

Comment (anonymouse64) [reply interfaces/builtin/opengl.go:118]: thanks looks good

**Review by anonymouse64 — APPROVED** [review]
+1

**Review by bboozzoo — APPROVED** [review]
LGTM. Do we need :+1: from security too?

File: interfaces/builtin/opengl.go [hunk 2]
```diff
@@ -114,6 +114,9 @@ unix (bind,listen) type=seqpacket addr="@cuda-uvmfd-[0-9a-f]*",
 # https://github.com/Xilinx/XRT/tree/master/src/runtime_src/core/edge/drm
 /sys/devices/platform/amba{,_pl@[0-9]*}/amba{,_pl@[0-9]*}:zyxclmm_drm/* r,
 
+# Imagination PowerVR driver
```
Comment (pedronis) [inline interfaces/builtin/opengl.go:117]: could you provide a link to some document from Imagination that at least lists this name?

Comment (EthanHsieh) [reply interfaces/builtin/opengl.go:117]: I didn't find any documents from Imagination official website. Have to register with imagination to get documents. 
Here is all information I can get: [1].  PowerVR GE8300[2] is the chip on MTK i300a board.
Not sure if [1] is enough for you.
[1] https://www.imaginationtech.com/powervr-gpu/ge8300/
[2] https://www.mediatek.com/products/AIoT/mt8362a
If it's okay to you, I'll send a PR to add [1] into opengl.go

Comment (pedronis) [reply interfaces/builtin/opengl.go:117]: no, it's fine. Did you test that this is all that is needed to make things work?

**Review by pedronis — APPROVED** [review]
I'm fine with this, assuming it's complete/enough.

File: interfaces/builtin/opengl.go (hunk 2 above)
Comment (EthanHsieh) [reply interfaces/builtin/opengl.go:117]: yes. I verified it by instructions[1] provided by mir team.
[1] https://bugs.launchpad.net/baoshan/+bug/1929173/comments/8

**Pull request merged**